src/sgame/botlib/bot_nav.cpp

```diff
@@ -264,6 +264,8 @@ void G_BotUpdatePath( int botClientNum, const botRouteTarget_t *target, botNavCm
 
 		cmd->havePath = !bot->needReplan;
 
+		bool overNavconStart = false;
```

Comment:
It is free to walk anywhere else. But it will enable the wall walk depending on the distance.

Reply:
Looks like it is to me. `moveOverOffmeshConnection` is called and if that succeeds, some variables are set to indicate that the bot is using an off-mesh connection. So I think it would make sense to store the information about the connection if and only if this succeeds.

Reply:
If you insist. But in that case, we do already have a problem, as the current code does exactly the same thing.

Reply:
> the current code does exactly the same thing.

In the current code I see this: 
```
	bool overNavcon = G_IsBotOverNavcon( selfClientNum );
...
	if ( overNavcon )
	{
		self->botMind->lastNavconTime = level.time;
	}
```

Which looks wrong. I'm suggesting to reset the navcon time only if the connection is actually used with moveOverOffmeshConnection.

Reply:
I did that.